src/_common/events/event-bus.service.ts

```diff
@@ -0,0 +1,49 @@
+import Vue from 'vue';
+
+export type EventBusDeregister = () => void;
+
+export class EventBus {
+	private e!: Vue;
+
+	private static global = new EventBus();
+
+	constructor() {
+		this.e = new Vue();
+	}
+
+	emit(event: string, ...args: any[]) {
+		this.e.$emit(event, ...args);
+	}
+
+	on(event: string, callback: Function): EventBusDeregister {
+		this.e.$on(event, callback);
+
+		return () => {
+			this.e.$off(event, callback);
+		};
+	}
+
+	once(event: string, callback: Function) {
+		this.e.$once(event, callback);
```

Comment:
shouldn't this technically also return a deregister func?

Reply:
I didn't change this bit in event bus. Actually, we don't even use once. I'm going to remove this.